src/main/java/edu/aau/groupc/canteenbackend/user/dto/UserDto.java

```diff
@@ -0,0 +1,35 @@
+package edu.aau.groupc.canteenbackend.user.dto;
+
+import edu.aau.groupc.canteenbackend.dto.DTO;
+import edu.aau.groupc.canteenbackend.user.User;
+import lombok.Data;
+import org.hibernate.validator.constraints.Length;
+
+import javax.validation.constraints.Pattern;
+import java.io.Serializable;
+
+@Data
+public class UserDto implements DTO, Serializable {
+
+    @Length(min = 3, message = "Username must be at least 3 characters long!")
+    @Length(max = 24, message = "Username cannot be longer than 24 characters!")
+    @Pattern(regexp = "^[A-Za-z0-9]+$", message = "Username can contain only alphanumeric characters!")
+    private String username;
```

Comment:
I think you need to add `@NotBlank(message = "Username is required")` to properties to prohibit null values. For some reason, `@Length` does on its own allow null values.
(Also applies to password and similar properties)

Reply:
That's strange behaviour. I have thought about it and didn't include it because I thought `@Length` is enough. Thanks for pointing out.